src/modules/Chat.ts

```diff
@@ -0,0 +1,44 @@
+import { User } from "src/modules/User";
+import { Socket } from "socket.io-client";
+
+interface chatMessage {
+  timestamp: number;
+  sender: string;
+  message: string;
+}
+export class Chat {
+  private user: User;
+  private roomId: string;
+  private channel: Socket;
+
+  constructor(user: User, roomId: string, channel: Socket) {
+    this.user = user;
+    this.roomId = roomId;
+    this.channel = channel;
+  }
+
+  sendMessage(message: string) {
+    this.channel.emit("postChatMessage", {
+      message,
+      roomId: this.roomId,
+      user: this.user?.getToken(),
```

Comment:
We need the token for this call, without it the user shouldn't be able to post the message.

Reply:
Token has no connection as to how user will postMessage that will be handled by socket Instance

Reply:
We have to send back the token to verify the user

Reply:
Can you add the suggested change? I still didn't get it.

Reply:
user: this.user.getToken(),